src/http.h

```diff
@@ -138,6 +145,14 @@ class HttpStateData : public Client
     static bool decideIfWeDoRanges (HttpRequest * orig_request);
     bool peerSupportsConnectionPinning() const;
 
+    /// Process an "101 Switching Protocols" reply.
+    /// \return false if this 101 reply is not supported, true otherwise
```

Comment:
> I may not know enough details to propose the best solution here, but Christos' description appears to imply that we should have two vectors instead of one:
`UpgradeProtocols *upgradeProtocolsReceivedFromClient = nullptr;`
`UpgradeProtocols *upgradeProtocolsSentToPeer = nullptr;`

There are two lists but we need to store only the second one. Even If we decide to delete upgradeProtocolsReceivedFromClient after build the upgradeProtocolsSentToPeer (as the comment says) we will have to do it inside the same function we built it and this is not normal.
The  upgradeProtocolsReceivedFromClient list is stored in received HttpRequest header if we need to access it in the future but currently there is not any reason to do it.
However If required and you insist we can add the upgradeProtocolsReceivedFromClient and mark that this is unused.

For now I renamed the upgradeProtocols to upgradeProtocolsSentToPeer.  
However in the code sometimes it should called  upgradeProtocolsToBeSentToPeer and sometimes upgradeProtocolsSentToPeer, in a point of view this is may make a confusion. 
This is why initially I selected the upgradeProtocols name.


> Document the true result. Positives are usually easier for the reader to comprehend.

Ok on this

Reply:
Both requests addressed at 41fc1c5a